go/tcp-service/server/main.go

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com/) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package main
+
+import (
+	"context"
+	"io"
+	"log"
+	"net"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+)
+
+// handles incoming client connection
+func handleConnection(conn net.Conn) {
+	defer conn.Close()
+	log.Printf("Accepted connection from %v", conn.RemoteAddr())
+
+	// Read data from the client
+	for {
+		buffer := make([]byte, 1024)
```

Comment:
We can move this outside the loop to avoid initializing it in each iteration. The conn.Read will overwrite the data inside the buffer.